src/main/java/org/rptest/core/logger/JavaUtilLogger.java

```diff
@@ -0,0 +1,31 @@
+package org.rptest.core.logger;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class JavaUtilLogger implements ILogger {
+    private final Logger logger;
+
+    public JavaUtilLogger() {
+        this.logger = Logger.getLogger(getCallerClassName());
+    }
+
+    public void info(String message) {
+        logger.log(Level.INFO, message);
+    }
+
+    public void error(String message) {
+        logger.log(Level.SEVERE, message);
+    }
+
+    private String getCallerClassName() {
+        StackTraceElement[] stElements = Thread.currentThread().getStackTrace();
+        String className = stElements[3].getClassName();
+        try {
+            return Class.forName(className).getName();
```

Comment:
Doesn't it will return same value as stored in 'className'?

Reply:
Good question. 

I tried with a few classes and it returned the same thing, but here's what chatGPT has to say about it:

_You're correct that in most cases, `stElements[3].getClassName();` and `Class.forName(className).getName();` will return the same value. The main reason for using `Class.forName(className)` in the try-catch block is to ensure that the class name obtained from the stack trace actually exists and can be loaded by the Java ClassLoader.

In some cases, the class might not be available at runtime due to various reasons such as class loading issues, classpath configuration problems, or the class being removed or relocated in the project. By using `Class.forName(className)`, the code checks if the class can be loaded successfully. If a `ClassNotFoundException` is thrown, it indicates that the class could not be found or loaded, and an error message is logged accordingly.

However, if you are confident that the class will always be available at runtime, you can simplify the method by removing the try-catch block and returning the class name directly from the stack trace element:_

```java
private String getCallerClassName() {
    StackTraceElement[] stElements = Thread.currentThread().getStackTrace();
    return stElements[3].getClassName();
}
```

_This simplified version of the method assumes that the class name obtained from the stack trace will always be valid and loadable. Keep in mind that if there's any issue with loading the class, this simplified version will not log an error message and will not handle the potential `ClassNotFoundException`._